src/modules/dart.rs

```diff
@@ -0,0 +1,80 @@
+use std::process::Command;
+
+use super::{Context, Module, RootModuleConfig, SegmentConfig};
+
+use crate::configs::dart::DartConfig;
+
+/// Creates a module with the current Dart version
+///
+/// Will display the Dart version if any of the following criteria are met:
+///     - Current directory contains a `.dart` file
+///     - Current directory contains a `pubspec.yaml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_dart_project = context
+        .try_begin_scan()?
+        .set_files(&["pubspec.yaml"])
+        .set_extensions(&["dart"])
+        .is_match();
+
+    if !is_dart_project {
+        return None;
+    }
+
+    match get_dart_version() {
+        Some(dart_version) => {
+            let mut module = context.new_module("dartlang");
+            let config: DartConfig = DartConfig::try_load(module.config);
+
+            module.set_style(config.style);
+            module.create_segment("symbol", &config.symbol);
+
+            let formatted_version = format_dart_version(&dart_version)?;
+            module.create_segment("version", &config.version.with_value(&formatted_version));
+
+            Some(module)
+        }
+        None => None,
+    }
+}
+
+fn get_dart_version() -> Option<String> {
+    match Command::new("dart").arg("--version").output() {
+        Ok(output) => {
+            if !output.status.success() {
+                log::warn!(
+                    "Non-Zero exit code '{}' when executing `dart --version`",
+                    output.status
+                );
+                return None;
+            }
+
+            Some(String::from_utf8(output.stderr).unwrap())
+        }
+        Err(_) => None,
+    }
+}
+
+fn format_dart_version(dart_stdout: &str) -> Option<String> {
```

Comment:
this function could look like this:
```rust
fn format_dart_version(dart_stdout: &str) -> Option<String> {
    // `dart --version` output will look somthing like this:
    // Dart VM version: 2.5.2 (Tue Apr 20 04:20:00 2019 +0200) on "macos_x64"

    let version = dart_stdout.split(' ').nth(3)?;

    Some(format!("v{}", version))
}
```
I think this way might be better than creating a new mutable String and appending to it.

Reply:
I totally stole that from the [Go module](https://github.com/starship/starship/blob/f2cb529d1bd44f996ad60c2d27ae14d1f9ad44ba/src/modules/golang.rs#L54-L69). Should we change that there too?

Reply:
I think this way makes a lot more sense but ultimately they both work.

Even though it's not the biggest of issues, I think I'll still open a PR for it. This way no one will copy that code again :)